CriteoPublisherSdk/Sources/Configuration/CR_IntegrationRegistry.m

```diff
@@ -80,4 +81,21 @@ - (BOOL)isAdMobMediationPresent {
          NSProtocolFromString(@"GADCustomEventBanner") != nil;
 }
 
+- (CR_CacheAdUnitArray *)filterAdUnits:(CR_CacheAdUnitArray *)units {
+  return [units
+      filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(id _Nullable adUnit,
+                                                                        NSDictionary<NSString *, id>
+                                                                            *_Nullable bindings) {
+        if ([adUnit adUnitType] != CRAdUnitTypeRewarded) {
+          return YES;
+        }
+        if ([self.profileId isEqual:@(CR_IntegrationGamAppBidding)]) {
+          return YES;
+        } else {
+          CRLogWarn(@"Bidding", @"RewardedAdUnit %@ can only be used with GAM app bidding",
```

Comment:
Do you mind adding a extra comment explaining the why of such filter ?

The why:
> OnlyGAM AppBidding is supporting rewarded ads because they are handling the display themselves.